### PR TITLE
{ceph,lvm}: generate new UUIDS for btrfs filesystems

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
@Mantwon, if you want to test this.

File: test/suites/storage.sh [hunk 1]
```diff
@@ -40,6 +40,20 @@ test_storage() {
...
+
+      # Test generation of unique UUID.
+      lxc init testimage uuid1 -s "lxdtest-$(basename "${LXD_DIR}")-pool-btrfs"
+      POOL="lxdtest-$(basename "${LXD_DIR}")-pool-btrfs"
+      lxc copy uuid1 uuid2
+      lxc start uuid1
+      lxc start uuid2
+      lxc stop --force uuid1
+      lxc stop --force uuid2
+      [ "$(blkid -s UUID -o value /dev/"${POOL}"/containers_uuid1)" != "$(blkid -s UUID -o value /dev/"${POOL}"/containers_uuid2)" ]
```
Comment (brauner) [inline test/suites/storage.sh]: I'm puzzled as to what is happening in these tests. Locally this works fine and for both btrfs and xfs new uuids are generated. Using `lsblk` will always show the same uuid which clearly is not the case. Using `blkid` will correctly show differing uuids in *some* cases and sometimes none at all. My only explanation is that somehow udev is not fast enough under heavy load. But I'm not sure.

Comment (stgraber) [reply test/suites/storage.sh]: What does udev have to do with this? isn't blkid just reading the partition header?

Comment (brauner) [reply test/suites/storage.sh]: It uses `libuuid` I think. Also, `libblkid` uses a cache.

Comment (brauner) [reply test/suites/storage.sh]: There might be a trick: `blkid --cache-file=/dev/null`

Comment (brauner) [reply test/suites/storage.sh]: `blkid` returns empty in a bunch of cases meaning it doesn't list the uuid.

**Comment by Mantwon** [comment]
Apology -- I could not get to this, my work is quite busy this time of year. 

feel free to ping me on future ones though as I am interested to help, things are starting to calm down for me now so should have more time.

**Comment by brauner** [comment]
@Mantwon, on no worries at all. I just remembered that you were using `ceph` + `btrfs` so I just wanted to let you know that this is probably a fix you want to include. :)

**Pull request merged**